data/scripts/talkactions/commands/attributes.lua

```diff
@@ -0,0 +1,322 @@
+        targetFunction = function(item, target)
+            return item:clone()
+        end
+    },
+
+    ["attack"] = {
+        isActive = true,
+        targetFunction = function(item, target)
+            return item:setAttribute(ITEM_ATTRIBUTE_ATTACK, target)
+        end
+    },
+
+    ["defense"] = {
+        isActive = true,
+        targetFunction = function(item, target)
+            return item:setAttribute(ITEM_ATTRIBUTE_DEFENSE, target)
+        end
+    },
+
+    ["extradefense"] = {
+        isActive = true,
+        targetFunction = function(item, target)
+            return item:setAttribute(ITEM_ATTRIBUTE_EXTRADEFENSE, target)
+        end
+    },
+
+    ["armor"] = {
+        isActive = true,
+        targetFunction = function(item, target)
+            return item:setAttribute(ITEM_ATTRIBUTE_ARMOR, target)
+        end
+    }
+}
+
+local creatureFunctions = {
+    ["health"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:addHealth(target)
+        end
+    },
+
+    ["mana"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:addMana(target)
+        end
+    },
+
+    ["speed"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:changeSpeed(target)
+        end
+    },
+
+    ["droploot"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:setDropLoot(target)
+        end
+    },
+
+    ["skull"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:setSkull(target)
+        end
+    },
+
+    ["direction"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:setDirection(target)
+        end
+    },
+
+    ["maxHealth"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            return creature:setMaxHealth(target)
+        end
+    },
+
+    ["say"] = {
+        isActive = true,
+        targetFunction = function(creature, target)
+            creature:say(target, TALKTYPE_SAY)
+        end
+    }
+}
+
+local playerFunctions = {
+    ["fyi"] = {
+        isActive = true,
+        targetFunction = function(player, target)
+            return player:popupFYI(target)
+        end
+    },
+
+    ["tutorial"] = {
+        isActive = true,
+        targetFunction = function(player, target)
+            return player:sendTutorial(target)
+        end
+    },
+
+    ["guildnick"] = {
+        isActive = true,
+        targetFunction = function(player, target)
+            return player:setGuildNick(target)
+        end
+    },
+
+    ["group"] = {
+        isActive = true,
+        targetFunction = function(player, target)
+            return player:setGroup(Group(target))
+        end
+    },
+
+    ["vocation"] = {
```

Comment:
This has a problem, it happens that when trying something like this: `/attr vocation,1` the parameter `1` is sent to function `Vocation` as a string and not as a number and then the result is: `[Warning - Vocations::getVocation] Vocation 65535 not found.`

This is not your fault, what happens is that in the sources the function: `isNumber` from the lua interface only checks if the value on the stack is really number and not anything convertible to a number.

Just modify the code and handle this case correctly ;)

Reply:
Could you submit a change suggestion?